core/src/test/java/com/datastax/oss/driver/internal/core/control/ControlConnectionEventsTest.java

```diff
@@ -46,16 +47,19 @@ public void should_register_for_all_events_if_topology_requested() {
 
     // When
     controlConnection.init(true, false, false);
-    waitForPendingAdminTasks();
-    DriverChannelOptions channelOptions = optionsCaptor.getValue();
 
     // Then
-    assertThat(channelOptions.eventTypes)
-        .containsExactly(
-            ProtocolConstants.EventType.SCHEMA_CHANGE,
-            ProtocolConstants.EventType.STATUS_CHANGE,
-            ProtocolConstants.EventType.TOPOLOGY_CHANGE);
-    assertThat(channelOptions.eventCallback).isEqualTo(controlConnection);
+    await()
+        .untilAsserted(
```

Comment:
Isn't `unitAsserted` too strict? There is an option that it will block indefinitely. I think it is better to have `wait().unit()` with some sane timeout.

Reply:
I think by default it times out after 10 seconds, like the other awaitility methods. Will check.

Reply:
Yes, it's 10 seconds:
```java
await().untilAsserted(() -> assertThat(true).isEqualTo(false));

org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a Tmp expected:<[fals]e> but was:<[tru]e> within 10 seconds.
```
But I also noticed that `untilAsserted` doesn't propertly throw a timeout exception in Awaitibility 3.1.6. I've upgraded it to 4.0.2.